From stable-bounces@linux.kernel.org Fri Jun  1 01:04:33 2007
From: Tim Gardner <tim.gardner@ubuntu.com>
Date: Fri, 01 Jun 2007 00:46:40 -0700
Subject: Work around Dell E520 BIOS reboot bug
To: torvalds@linux-foundation.org
Cc: tim.gardner@ubuntu.com, akpm@linux-foundation.org, Matt_Domsch@dell.com, stable@kernel.org, ak@suse.de
Message-ID: <200706010746.l517kdxp005490@shell0.pdx.osdl.net>


From: Tim Gardner <tim.gardner@ubuntu.com>

Force Dell E520 to use the BIOS to shutdown/reboot.

I have at least one report that this patch fixes shutdown/reboot
problems on the Dell E520 platform.

(Andi says: People can always set the boot option.  It hardly seems like a
critical issue needing a backport.)

Signed-off-by: Tim Gardner <tim.gardner@ubuntu.com>
Acked-by: Andi Kleen <ak@suse.de>
Acked-by: Matt Domsch <Matt_Domsch@dell.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>


---
 arch/i386/kernel/reboot.c |    8 ++++++++
 1 file changed, 8 insertions(+)

--- linux-2.6.21.4.orig/arch/i386/kernel/reboot.c
+++ linux-2.6.21.4/arch/i386/kernel/reboot.c
@@ -88,6 +88,14 @@ static int __init set_bios_reboot(struct
 }
 
 static struct dmi_system_id __initdata reboot_dmi_table[] = {
+	{	/* Handle problems with rebooting on Dell E520's */
+		.callback = set_bios_reboot,
+		.ident = "Dell E520",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Dell Inc."),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Dell DM061"),
+		},
+	},
 	{	/* Handle problems with rebooting on Dell 1300's */
 		.callback = set_bios_reboot,
 		.ident = "Dell PowerEdge 1300",
